pkg/tmpl/context_funcs_test.go

```diff
@@ -82,6 +84,94 @@ func TestReadFile(t *testing.T) {
 	}
 }
 
+type entry struct {
+	name  string
+	fType fs.FileMode
+	isDir bool
+}
+
+func (e *entry) Name() string {
+	return e.name
+}
+
+func (e *entry) IsDir() bool {
+	return e.isDir
+}
+
+func (e *entry) Type() fs.FileMode {
+	return e.fType
+}
+
+func (e *entry) Info() (fs.FileInfo, error) {
+	return nil, fmt.Errorf("You should not call this method")
+}
+
+func TestReadDir(t *testing.T) {
+	result := []fs.DirEntry{
+		&entry{name: "file1.yaml"},
+		&entry{name: "file2.yaml"},
+		&entry{name: "file3.yaml"},
+		&entry{name: "folder1", isDir: true},
+	}
+	expectedArrayWindows := []string{
+		"sampleDirectory\\file1.yaml",
+		"sampleDirectory\\file2.yaml",
+		"sampleDirectory\\file3.yaml",
+	}
+	expectedArrayUnix := []string{
+		"sampleDirectory/file1.yaml",
+		"sampleDirectory/file2.yaml",
+		"sampleDirectory/file3.yaml",
+	}
+	var expectedArray []string
+	if runtime.GOOS == "windows" {
+		expectedArray = expectedArrayWindows
+	} else {
+		expectedArray = expectedArrayUnix
+	}
+
+	expectedDirname := "sampleDirectory"
+	ctx := &Context{basePath: ".", readDir: func(dirname string) ([]fs.DirEntry, error) {
+		if dirname != expectedDirname {
+			return nil, fmt.Errorf("unexpected filename: expected=%v, actual=%s", expectedDirname, dirname)
+		}
+		return result, nil
+	}}
+
+	actual, err := ctx.ReadDir(expectedDirname)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	if !reflect.DeepEqual(actual, expectedArray) {
+		t.Errorf("unexpected result: expected=%v, actual=%v", expectedArray, actual)
+	}
+}
+
+func TestReadDirEntries(t *testing.T) {
+	result := []fs.DirEntry{
+		&entry{name: "file1.yaml"},
+		&entry{name: "file2.yaml"},
+		&entry{name: "file3.yaml"},
+		&entry{name: "folder1", isDir: true},
+	}
+
+	expectedDirname := "sampleDirectory"
+	ctx := &Context{basePath: ".", readDir: func(dirname string) ([]fs.DirEntry, error) {
+		if dirname != expectedDirname {
```

Comment:
@vasicvuk Please try to use `requires` library. Thanks very much

Reply:
@yxxhero  Sorry, i don't understand. I see that there are require calls for test cooperation but this is a mock of a function, not a test case?